src/main/java/software/amazon/awssdk/crt/http/HttpException.java

```diff
@@ -0,0 +1,46 @@
+/*
+ * Copyright 2010-2019 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License").
+ * You may not use this file except in compliance with the License.
+ * A copy of the License is located at
+ *
+ *  http://aws.amazon.com/apache2.0
+ *
+ * or in the "license" file accompanying this file. This file is distributed
+ * on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either
+ * express or implied. See the License for the specific language governing
+ * permissions and limitations under the License.
+ */
+
+package software.amazon.awssdk.crt.http;
+
+import software.amazon.awssdk.crt.CRT;
+
+/**
+ * This exception will be thrown by any exceptional cases encountered within the
+ * JNI bindings to the AWS Common Runtime
+ */
+public class HttpException extends RuntimeException {
+    private static final int DEFAULT_ERROR_CODE = -1;
+    private int errorCode;
+
+    public HttpException(String msg) {
+        super(msg);
+        this.errorCode = DEFAULT_ERROR_CODE;
```

Comment:
Don't have to do this now, but it wouldn't hurt to resolve the error code to a string to make the exception message helpful. See ```CRT.awsErrorString()```

Reply:
Removed this String constuctor, leaving only one that takes the CRT ErrorCode and resolves the Error string using `CRT.awsErrorString()`.